pkg/kubelet/volume/gce_pd/gce_pd.go

```diff
@@ -194,9 +194,31 @@ func (pd *gcePersistentDisk) SetUp() error {
 	globalPDPath := makeGlobalPDName(pd.plugin.host, pd.pdName, pd.readOnly)
 	err = pd.mounter.Mount(globalPDPath, pd.GetPath(), "", mount.FlagBind|flags, "")
 	if err != nil {
-		os.RemoveAll(pd.GetPath())
-		// TODO: we should really eject the attach/detach out into its own control loop.
-		detachDiskLogError(pd)
+		mountpoint, mntErr := isMountPoint(pd.GetPath())
+		if mntErr != nil {
+			glog.Errorf("isMountpoint check failed: %v", mntErr)
+			return err
+		}
+		if mountpoint {
+			if mntErr = pd.mounter.Unmount(pd.GetPath(), 0); mntErr != nil {
+				glog.Errorf("Failed to unmount: %v", mntErr)
+				return err
+			}
+			mountpoint, mntErr := isMountPoint(pd.GetPath())
+			if mntErr != nil {
+				glog.Errorf("isMountpoint check failed: %v", mntErr)
+				return err
+			}
+			if mountpoint {
+				// This is very odd, we don't expect it.  We'll try again next sync loop.
+				glog.Errorf("%s is still mounted, despite call to unmount().  Will try again next sync loop.", pd.GetPath())
+				return err
+			}
+		} else {
```

Comment:
This should not be an else, just move to to parent scope.  The only way to arrive here is if isMountPoint() returned false.